test/hotspot/jtreg/vmTestbase/nsk/jvmti/GetObjectMonitorUsage/objmonusage003.java

```diff
@@ -28,9 +28,9 @@
 public class objmonusage003 {
 
     final static int JCK_STATUS_BASE = 95;
-    final static int NUMBER_OF_THREADS = 16;
-    final static int WAIT_TIME = 100;
-    volatile static boolean waiterInLockCheck = false;
+    final static int NUMBER_OF_ENTERER_THREADS = 4;
+    final static int NUMBER_OF_WAITER_THREADS  = 4;
+    final static int NUMBER_OF_THREADS = NUMBER_OF_ENTERER_THREADS + NUMBER_OF_WAITER_THREADS;
```

Comment:
This testing is not sufficient. We have three states of interest:
- entering the monitor directly
- waiting in the monitor via Object.wait
- re-rentering the monitor after being notified (or timed-out or interrupted)
- we need to check all of these conditions in permutations covering zero and non-zero for each one, and then see that we get the correct counts back from JVM TI.

Reply:
The last suggestion has been addressed by adding missing test cases to the `nsk/jvmti/GetObjectMonitorUsage/objmonusage003` test.